packages/clerk-js/src/ui/utils/passwordUtils.ts

```diff
@@ -0,0 +1,72 @@
+import type { ClerkAPIError, PasswordSettingsData } from '@clerk/types';
+
+import type { LocalizationKey } from '../localization';
+import { localizationKeys } from '../localization/localizationKeys';
+import { canUseListFormat } from '../utils';
+
+// match FAPI error codes with localization keys
+export const mapComplexityErrors = (passwordSettings: Pick<PasswordSettingsData, 'max_length' | 'min_length'>) => {
+  return {
+    form_password_length_too_long: [
+      'unstable__errors.passwordComplexity.maximumLength',
+      'length',
+      passwordSettings.max_length,
+    ],
+    form_password_length_too_short: [
+      'unstable__errors.passwordComplexity.minimumLength',
+      'length',
+      passwordSettings.min_length,
+    ],
+    form_password_no_uppercase: 'unstable__errors.passwordComplexity.requireUppercase',
+    form_password_no_lowercase: 'unstable__errors.passwordComplexity.requireLowercase',
+    form_password_no_number: 'unstable__errors.passwordComplexity.requireNumbers',
+    form_password_no_special_char: 'unstable__errors.passwordComplexity.requireSpecialCharacter',
+  };
+};
+
+type LocalizationConfigProps = {
+  t: (localizationKey: LocalizationKey | string | undefined) => string;
+  locale: string;
+  passwordSettings: Pick<PasswordSettingsData, 'max_length' | 'min_length'>;
+};
+export const createPasswordError = (errors: ClerkAPIError[], localizationConfig: LocalizationConfigProps) => {
+  if (!localizationConfig) {
+    return errors[0].longMessage;
+  }
+
+  const { t, locale, passwordSettings } = localizationConfig;
+
+  const message = errors.map((s: any) => {
```

Comment:
❓ There is a similar mapping logic inside `generateErrorTextUtil`. Is this something we can merge here as well or it is not possible ?

Reply:
@panteliselef I looked into this and it is slightly different. Merging would make it a bit verbose for no real benefit so I chose to keep them separate.